src/sentry/incidents/logic.py

```diff
@@ -735,6 +754,67 @@ def snapshot_alert_rule(alert_rule, user=None):
     )
 
 
+def should_update_seer(
+    previous_detection_type: AlertRuleDetectionType,
+    updated_detection_type: AlertRuleDetectionType,
+    query: str,
+    aggregate: str,
+) -> bool:
+    """
+    Update Seer data if a dynamic alert's query or aggregate has changed OR
+    a static or percent alert has changed to become a dynamic alert
+    """
+    if (
+        previous_detection_type == AlertRuleDetectionType.DYNAMIC
+        and updated_detection_type == AlertRuleDetectionType.DYNAMIC
+    ):
+        if query or aggregate:
+            return True
+    elif updated_detection_type == AlertRuleDetectionType.DYNAMIC:
+        return True
+    return False
+
+
+def update_dynamic_alert_data(
+    alert_rule: AlertRule,
+    seasonality: AlertRuleSeasonality,
+    sensitivity: AlertRuleSensitivity,
+    threshold_type: int,
+    query: str,
+    time_window: int,
+    dataset: str,
+    aggregate: str,
+    projects: list[Project],
+) -> None:
+    updated_seasonality = seasonality if seasonality is not NOT_SET else alert_rule.seasonality
+    updated_sensitivity = sensitivity if sensitivity is not NOT_SET else alert_rule.sensitivity
+
+    rule_data = DynamicAlertRuleData(
+        alert_rule_id=alert_rule.id,
+        seasonality=updated_seasonality.value,
+        sensitivity=updated_sensitivity.value,
+        threshold_type=threshold_type or alert_rule.threshold_type,
+    )
+    snuba_query_data = SnubaQueryData(
+        query=query or alert_rule.snuba_query.query,
+        time_window=time_window or alert_rule.snuba_query.time_window,
+        dataset=dataset or alert_rule.snuba_query.dataset,
+        aggregate=aggregate or alert_rule.snuba_query.aggregate,
+    )
+    project = projects[0] if projects else alert_rule.projects.get()
+    # do we need to tell seer when sensitivity, seasonality, or threshold_type changes?
+    # or can they get that from the periodic updates?
```

Comment:
What are periodic updates? Does this mean from the subscription processor ingest?

Ideally we be explicit with the updates - when a customer updates the model, we propagate that through to Seer

Reply:
yeah just the regular metric alert updates